src/api/api.ts

```diff
@@ -121,7 +122,7 @@ export const getAudiences = async (client: IDeskproClient, email?: string): Prom
   }
 }
 
-export const updateAudienceSubscription = async (client: IDeskproClient, audienceId: string, email: string, status: AudienceStatus): Promise<void> => {
+export const updateAudienceSubscription = async (client: IDeskproClient, audienceId: string, email: string, status: AudienceStatus): Promise<boolean|string[]> => {
```

Comment:
This is a kind of confusing return type. I might suggest either returning something more structured, like a discriminated union of possible results, or throw with a custom error for your error cases.